modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/util.rb

```diff
@@ -69,10 +69,12 @@ def using_user_token(storage, user, &)
                 )
             end
 
-            def storage_error(response:, code:, source:)
-              data = StorageErrorData.new(source:, payload: response.json(symbolize_keys: true))
+            def storage_error(response:, code:, source:, log_message: nil)
+              # Some errors, like timeouts, aren't json responses so we need to adapt
+              payload = response.respond_to?(:json) ? response.json(symbolize_keys: true) : response.to_s
```

Comment:
🟡 rather then using `respond_to?(:json)`, can we use `response.content_type.mime_type`? in the nextcloud util we have such a matching: 

```ruby
            def error_data_from_response(caller:, response:)
              payload =
                case response.content_type.mime_type
                when "application/json"
                  response.json
                when "text/xml", "application/xml"
                  response.xml
                else
                  response.body.to_s
                end

              StorageErrorData.new(source: caller, payload:)
            end
```

Reply:
It doesn't seem to be the case as we have 2 things in play here: 

1. a `Response` that happens to be an error to us
2. an `HTTPXErrorResponse` which happens when the request wasn't able to be made at all

So the first one we could parse the mime_type (in theory should just be json as this is only for the MS API) but on the second one not really.

Reply:
I think, the current approach is alright. In the end, it is just a subjective point of view.